extras/ams_wrapper/src/api/models/model.py

```diff
@@ -63,38 +64,64 @@ def postprocess_inference_output(self, inference_output: dict) -> str:
     def on_post(self, req, resp):
         # Main flow for the inference request
 
-        # TODO: Handle errors
-
-        # Retrieve request headers as python dictionary 
+        # Retrieve request headers as python dictionary
         request_headers = req.headers
         logger.debug(f"Received request with headers: {request_headers}")
         # Retrieve raw bytes from the request
         request_body = req.bounded_stream.read()
+        inference_output = {}
 
         # Preprocess request body
-        preprocessing_start_time = datetime.datetime.now()
-        input_image = self.preprocess_binary_image(request_body)
-        duration = (datetime.datetime.now() -
-                    preprocessing_start_time).total_seconds() * 1000
-        logger.debug(f"Input preprocessing time: {duration} ms")
+        try:
+            preprocessing_start_time = datetime.datetime.now()
+            input_image = self.preprocess_binary_image(request_body)
+            duration = (datetime.datetime.now() -
+                        preprocessing_start_time).total_seconds() * 1000
+            logger.debug(f"Input preprocessing time: {duration} ms")
+        except Exception as ex:
+            body = {"message": str(ex)}
+            resp.status = falcon.HTTP_400
+            resp.body = json.dumps(body)
+            return
 
         # Send inference request to corresponding model in OVMS
-        connection_start_time = datetime.datetime.now()
-        inference_ouput = self.ovms_connector.send(input_image)
-        duration = (datetime.datetime.now() -
-                    connection_start_time).total_seconds() * 1000
-        logger.debug(f"OVMS request handling time: {duration} ms")
+        try:
+            connection_start_time = datetime.datetime.now()
+            inference_ouput = self.ovms_connector.send(input_image)
+            duration = (datetime.datetime.now() -
+                        connection_start_time).total_seconds() * 1000
+            logger.debug(f"OVMS request handling time: {duration} ms")
+        except OvmsUnavailableError as ex:
+            body = {"message": str(ex)}
```

Comment:
I'd add `logger.exception()` call here, so full stack trace will be visible in the logs (perhaps it would be helpful to log exceptions in other try except blocks here).